File: secure_token.py
#!/usr/bin/env python3

import base64
import hashlib

secret = 'mysecret'
uri = '/my/uri'
expire = '1470055000'

md5 = hashlib.md5()
md5.update((secret + uri + expire).encode('utf-8'))
token = base64.b64encode(md5.digest()).decode('utf-8').replace('=','').replace('+','-').replace('/','_')

print('http://www.example.org%s?st=%s&e=%s' % (uri, token, expire))

File: secure_token.js
const crypto = require('crypto');

const secret = 'mysecret';
const uri = '/my/uri';
const expire = '1470055000';

const md5 = crypto.createHash('md5');
md5.update((secret + uri + expire), 'utf-8');
const token = Buffer.from(md5.digest()).toString('base64').replace(/=|\+|\//g, (match) => ({
  '=': '',
  '+': '-',
  '/': '_',
}[match]));

console.log(`http://www.example.org${uri}?st=${token}&e=${expire}`);

